### PR TITLE
Switch to scikit-build-core

### PR DESCRIPTION
**Pull request by vyasr**
Contributes to rapidsai/build-planning#2

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -190,11 +190,15 @@ if hasArg --pydevelop; then
...
-# Append `-DFIND_RAFT_CPP=ON` to EXTRA_CMAKE_ARGS unless a user specified the option.
 SKBUILD_EXTRA_CMAKE_ARGS="${EXTRA_CMAKE_ARGS}"
-  if [[ "${EXTRA_CMAKE_ARGS}" != *"DFIND_CUGRAPH_CPP"* ]]; then
-      SKBUILD_EXTRA_CMAKE_ARGS="${SKBUILD_EXTRA_CMAKE_ARGS} -DFIND_CUGRAPH_CPP=ON"
-  fi
+
+# Replace spaces with semicolons in SKBUILD_EXTRA_CMAKE_ARGS
+SKBUILD_EXTRA_CMAKE_ARGS=$(echo ${SKBUILD_EXTRA_CMAKE_ARGS} | sed 's/ /;/g')
+
+# Append `-DFIND_RAFT_CPP=ON` to EXTRA_CMAKE_ARGS unless a user specified the option.
```
Comment (bdice) [inline build.sh:198]:
```suggestion
# Append `-DFIND_CUGRAPH_CPP=ON` to EXTRA_CMAKE_ARGS unless a user specified the option.
```

**Review by bdice — APPROVED** [review]
Approving with one small fix suggestion.

**Review by ajschmidt8 — APPROVED** [review]

File: build.sh [hunk 2]
```diff
@@ -333,8 +337,7 @@ if buildDefault || hasArg pylibcugraph || hasArg all; then
                    -j${PARALLEL_LEVEL:-1}
             cd -
         fi
-        SKBUILD_CONFIGURE_OPTIONS="${SKBUILD_EXTRA_CMAKE_ARGS} -DUSE_CUGRAPH_OPS=${BUILD_WITH_CUGRAPHOPS}" \
-            SKBUILD_BUILD_OPTIONS="-j${PARALLEL_LEVEL}" \
+        SKBUILD_CMAKE_ARGS="${SKBUILD_EXTRA_CMAKE_ARGS};-DUSE_CUGRAPH_OPS=${BUILD_WITH_CUGRAPHOPS}" \
```
Comment (rlratzel) [inline build.sh:340]: I can't comment on lines too far outside the change, but can the `FIXME` just above (starting on line 324) and below (starting on line 350) be removed now?

**Review by rlratzel — COMMENTED** [review]
Nothing jumped out so it LGTM.  I do have a question about some `FIXME`s though...

**Review by rlratzel — APPROVED** [review]
Excellent, it's great to be able to remove that FIXME special case, thanks!

**Pull request merged**